WooCommerce/Classes/ViewRelated/Orders/OrderStatusListViewController.swift

```diff
@@ -0,0 +1,249 @@
+    ///
+    private func registerTableViewCells() {
+        let cells = [StatusListTableViewCell.self]
+
+        for cell in cells {
+            tableView.register(cell.loadNib(), forCellReuseIdentifier: cell.reuseIdentifier)
+        }
+    }
+
+    /// Setup: TableView
+    ///
+    func configureTableView() {
+        view.backgroundColor = StyleManager.tableViewBackgroundColor
+        tableView.backgroundColor = StyleManager.tableViewBackgroundColor
+        tableView.refreshControl = refreshControl
+
+        tableView.dataSource = self
+        tableView.delegate = self
+    }
+
+    @IBAction func pullToRefresh(sender: UIRefreshControl) {
+        reload {
+            sender.endRefreshing()
+        }
+    }
+
+    private func reload(completion: () -> Void) {
+        configureResultsController()
+        completion()
+    }
+}
+
+/// MARK: - Navigation bar
+///
+extension OrderStatusListViewController {
+    func configureNavigationBar() {
+        configureTitle()
+        configureLeftButton()
+        configureRightButton()
+    }
+
+    func configureTitle() {
+        title = NSLocalizedString("Order Status", comment: "Change order status screen - Screen title")
+    }
+
+    func configureLeftButton() {
+        let dismissButtonTitle = NSLocalizedString("Cancel",
+                                                   comment: "Change order status screen - button title for closing the view")
+        let leftBarButton = UIBarButtonItem(title: dismissButtonTitle,
+                                            style: .plain,
+                                            target: self,
+                                            action: #selector(dismissButtonTapped))
+        leftBarButton.tintColor = .white
+        navigationItem.setLeftBarButton(leftBarButton, animated: false)
+    }
+
+    func configureRightButton() {
+        let applyButtonTitle = NSLocalizedString("Apply",
+                                               comment: "Change order status screen - button title to apply selection")
+        let rightBarButton = UIBarButtonItem(title: applyButtonTitle,
+                                             style: .done,
+                                             target: self,
+                                             action: #selector(applyButtonTapped))
+        rightBarButton.tintColor = .white
+        navigationItem.setRightBarButton(rightBarButton, animated: false)
+        deActivateApplyButton()
+    }
+
+    func activateApplyButton() {
+        navigationItem.rightBarButtonItem?.isEnabled = true
+    }
+
+    func deActivateApplyButton() {
+        navigationItem.rightBarButtonItem?.isEnabled = false
+    }
+
+    @objc func dismissButtonTapped() {
+        dismiss(animated: true, completion: nil)
+    }
+
+    @objc func applyButtonTapped() {
+        updateOrderStatus()
+        dismiss(animated: true, completion: nil)
+    }
+}
+
+
+/// MARK: - Update the Order status
+///
```

Comment:
Just a style nit — you may want to change the `/// MARK: - ...` comments to 

```
// MARK: - Update the Order status
//
```

So we get the nice seporators in the XCode pull-down:

<img width="334" alt="screen shot 2019-03-07 at 12 10 10 pm" src="https://user-images.githubusercontent.com/154014/53978990-318aca80-40d2-11e9-93a9-76f486a5a173.png">